### PR TITLE
feat(a11y): improve accessibility for the SnackBar component

### PR DESCRIPTION
**Pull request by felixzapata**
+ set the focus when the message is opened.
+ return the focus to the element which has opened the message.
+ show focus style when you focus over the button inside the message.

Related with #225.

### CODE REVIEW

File: components/snackbar/Snackbar.js [hunk 1]
```diff
@@ -39,15 +40,31 @@ const factory = (Overlay, Button) => {
     }
 
     componentWillReceiveProps (nextProps) {
+      const listenerTransition = () => {
+        this.setFocus();
+        this.refs.snackbar.removeEventListener('transitionend', listenerTransition);
+      };
+      if (nextProps.active) {
+        this.refs.snackbar.addEventListener('transitionend', listenerTransition);
```
Comment (javivelasco) [inline components/snackbar/Snackbar.js]: Isn't this adding tons event listeners? I think it should be checked if the listener is already added. May you look into it?

**Comment by felixzapata** [comment]
ok, I will review it

File: components/snackbar/Snackbar.js (hunk 1 above)
Comment (felixzapata) [inline components/snackbar/Snackbar.js]: ok, I will review it

Comment (SleeplessByte) [inline components/snackbar/Snackbar.js]: Instead, wouldn't it be better to add to `componentDidMount` and `componentWillUnmount`, register and unregister once, and simply handle it in the callback using props?.

**Pull request closed without merging**